test/diff.jl

```diff
@@ -370,4 +370,15 @@ let
     @test Symbolics.is_derivative(Symbolics.unwrap(D(X + 2a*Y)))
     @test !Symbolics.is_derivative(Symbolics.unwrap(D(X) + D(Y)))
     @test !Symbolics.is_derivative(Symbolics.unwrap(my_f(X, D(Y))))
+end
+
+# Zeroth derivative (#1163)
+let
+    @variables t X(t) 
+    Dt = Differential(t)^0
+    DXt = Differential(X)^0 
+    @test isequal(Dt, t)
```

Comment:
Oops, you're right. I got the meaning of `Differential` confused in my head. It shoudl probably be `identity`, right? So that e.g.

```julia
julia> D = Differential(t)^0
identity (generic function with 1 method)

julia> D(t + 2t^2)
t + 2(t^2)
```